streams/test-utils/src/main/java/org/apache/kafka/streams/internals/KeyValueStoreFacade.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.internals;
+
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.TimestampedKeyValueStore;
+import org.apache.kafka.streams.state.ValueAndTimestamp;
+import org.apache.kafka.streams.state.internals.ReadOnlyKeyValueStoreFacade;
+
+import java.util.List;
+
+public class KeyValueStoreFacade<K, V> extends ReadOnlyKeyValueStoreFacade<K, V> implements KeyValueStore<K, V> {
+
+    public KeyValueStoreFacade(final TimestampedKeyValueStore<K, V> inner) {
+        super(inner);
+    }
+
+    @Override
+    public void init(final ProcessorContext context,
+                     final StateStore root) {
+        inner.init(context, root);
+    }
+
+    @Override
+    public void put(final K key,
+                    final V value) {
+        inner.put(key, value, -1L);
+    }
+
+    @Override
+    public V putIfAbsent(final K key,
+                         final V value) {
+        final ValueAndTimestamp<V> old = inner.putIfAbsent(key, value, -1L);
+        return old == null ? null : old.value();
+    }
+
+    @Override
+    public void putAll(final List<KeyValue<K, V>> entries) {
+        for (final KeyValue<K, V> entry : entries) {
```

Comment:
Should we prepare a new list of key-valuetimstamp and call the inner.putAll here?

Reply:
This is more efficient. If we prepare a new list, we iterator over the list effectively twice (once here, and a second time in `inner.putAll()`). And this code is simple/straight forward, so I don't think we gain reduced code complexity / code reuse that justifies this overhead.